app/src/main/java/com/duckduckgo/app/bookmarks/ui/BookmarksViewModel.kt

```diff
@@ -339,4 +353,26 @@ class BookmarksViewModel @Inject constructor(
             )
         }
     }
+
+    fun onFaviconsFetchingEnabled(
+        fetchingEnabled: Boolean,
+        currentFolderId: String,
+    ) {
+        viewModelScope.launch(dispatcherProvider.io()) {
+            faviconsFetchingPrompt.onPromptAnswered(fetchingEnabled)
+        }
+        if (fetchingEnabled) {
+            refreshBookmarks(currentFolderId)
+        }
+    }
+
+    private fun refreshBookmarks(currentFolderId: String) {
```

Comment:
this method/intention is not clear. It seems we want to refresh bookmarks, but I think this is somehow related to the favicons.

I see we call it after fetchingEnabled. My assumption is to force some redraw. Instead why not a command that is more explicit about the action we want to perform. And then you do in the adapter what is required? that way the intention is more clear, and the required logic close to the component that renders. Otherwise it's not clear why we are refreshing.